node/worker/worker.go

```diff
@@ -112,14 +112,20 @@ func (w *Worker) CommitTransactions(txs types.Transactions, coinbase common.Addr
 	return nil
 }
 
-// CommitReceipts commits a list of receipts.
-func (w *Worker) CommitReceipts(receiptsList []types.CXReceipts, coinbase common.Address) error {
+// CommitReceipts commits a list of already verified incoming cross shard receipts
+func (w *Worker) CommitReceipts(receiptsList []*types.CXReceiptsProof, coinbase common.Address) error {
 	if w.current.gasPool == nil {
 		w.current.gasPool = new(core.GasPool).AddGas(w.current.header.GasLimit)
 	}
-	for _, receipts := range receiptsList {
-		// TODO: apply receipt
-		_ = receipts
+
+	if len(receiptsList) == 0 {
+		w.current.header.IncomingReceiptHash = types.EmptyRootHash
+	} else {
+		w.current.header.IncomingReceiptHash = types.DeriveSha(types.CXReceiptsProofs(receiptsList))
+	}
+
```

Comment:
You need to apply (add balance) with the receipts too. Take a look at the commitTransactions.

Reply:
Good catch. I added. Still got the same error